aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -440,38 +442,50 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
+      TableIdentifier identifier, String metadataFileLocation) {
+    Preconditions.checkArgument(
+        isValidIdentifier(identifier), "Table identifier to register is invalid: " + identifier);
+    Preconditions.checkArgument(
+        metadataFileLocation != null && !metadataFileLocation.isEmpty(),
         "Cannot register an empty metadata file location as a table");
 
-    Map<String, String> tableParameters = ImmutableMap.of(
-        BaseMetastoreTableOperations.TABLE_TYPE_PROP,
-        BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toLowerCase(Locale.ENGLISH),
-        BaseMetastoreTableOperations.METADATA_LOCATION_PROP,
-        metadataFileLocation);
+    TableOperations ops = newTableOps(identifier);
+    InputFile metadataFile = ops.io().newInputFile(metadataFileLocation);
+    TableMetadata metadata = TableMetadataParser.read(ops.io(), metadataFile);
 
-    TableInput tableInput = TableInput.builder()
-        .name(IcebergToGlueConverter.getTableName(identifier, awsProperties.glueCatalogSkipNameValidation()))
-        .tableType(GlueTableOperations.GLUE_EXTERNAL_TABLE_TYPE)
-        .parameters(tableParameters)
-        .build();
+    Map<String, String> tableParameters =
+        ImmutableMap.of(
+            BaseMetastoreTableOperations.TABLE_TYPE_PROP,
+            BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toLowerCase(Locale.ENGLISH),
+            BaseMetastoreTableOperations.METADATA_LOCATION_PROP,
+            metadataFileLocation);
+
+    String databaseName =
+        IcebergToGlueConverter.getDatabaseName(
+            identifier, awsProperties.glueCatalogSkipNameValidation());
+    String tableName =
+        IcebergToGlueConverter.getTableName(
+            identifier, awsProperties.glueCatalogSkipNameValidation());
+
+    TableInput tableInput =
+        TableInput.builder()
+            .applyMutation(
+                builder -> IcebergToGlueConverter.setTableInputInformation(builder, metadata))
+            .name(tableName)
+            .tableType(GlueTableOperations.GLUE_EXTERNAL_TABLE_TYPE)
+            .parameters(tableParameters)
+            .build();
 
     try {
-      glue.createTable(CreateTableRequest.builder()
-          .databaseName(IcebergToGlueConverter.getDatabaseName(identifier,
-              awsProperties.glueCatalogSkipNameValidation()))
-          .tableInput(tableInput)
-          .build());
+      glue.createTable(
+          CreateTableRequest.builder().databaseName(databaseName).tableInput(tableInput).build());
     } catch (software.amazon.awssdk.services.glue.model.AlreadyExistsException e) {
-      glue.updateTable(UpdateTableRequest.builder()
-          .databaseName(IcebergToGlueConverter.getDatabaseName(identifier,
-              awsProperties.glueCatalogSkipNameValidation()))
-          .tableInput(tableInput)
-          .build());
+      glue.updateTable(
+          UpdateTableRequest.builder().databaseName(databaseName).tableInput(tableInput).build());
```

Comment:
I think we need to first get the table to get the version number so it can be an input for update table